src/data.rs

```diff
@@ -106,6 +106,60 @@ impl Fields {
     }
 }
 
+#[cfg(any(feature = "full", feature = "derive"))]
+mod iter_member {
+    use super::*;
+    use crate::Member;
+
+    impl Fields {
+        /// Get an iterator over the fields of a struct or variant as [`Member`]s.
+        /// This iterator can be used to iterate over a named or unnamed struct or
+        /// variant's fields uniformly.
+        ///
+        /// The return type can considered as impl [`Iterator<Item = Member>`].
+        #[cfg_attr(docsrs, doc(cfg(any(feature = "full", feature = "derive"))))]
+        pub fn iter_member(&self) -> IterMember {
+            IterMember {
+                iter: self.iter(),
+                unnamed_counter: 0,
+            }
+        }
+    }
+
+    /// An iterator over the fields of a struct or variant as [`Member`]s.
+    #[cfg_attr(docsrs, doc(cfg(any(feature = "full", feature = "derive"))))]
+    #[derive(Clone)]
+    pub struct IterMember<'a> {
```

Comment:
this type should be publicly accessible ... so probably don't put it in its own private module

Reply:
I have also considered this issue before, but I noticed that many [iterator function](https://docs.rs/syn/latest/syn/struct.Generics.html#method.lifetimes) s under `Generics` also return private types. What do you think?